lms/static/js/learner_dashboard/views/entitlement_unenrollment_view.js

```diff
@@ -0,0 +1,149 @@
+                refundInfoSelector: '.js-entitlement-unenrollment-modal .js-entitlement-unenrollment-modal-refund-info',
+                errorInfoSelector: '.js-entitlement-unenrollment-modal .js-entitlement-unenrollment-modal-error-info',
+                submitButtonSelector: '.js-entitlement-unenrollment-modal .js-entitlement-unenrollment-modal-submit',
+                triggerSelector: '.js-entitlement-action-unenroll',
+                mainPageSelector: '#dashboard-main',
+
+                initialize: function(options) {
+                    var view = this;
+                    this.dashboardUrl = options.dashboardUrl;
+
+                    this.$submitButton = $(this.submitButtonSelector);
+                    this.$trackInfo = $(this.trackInfoSelector);
+                    this.$refundInfo = $(this.refundInfoSelector);
+                    this.$errorInfo = $(this.errorInfoSelector);
+
+                    this.$submitButton.on('click', this.handleSubmit.bind(this));
+
+                    $(this.triggerSelector).each(function() {
+                        var $trigger = $(this);
+
+                        $trigger.on('click', view.handleTrigger.bind(view));
+
+                        if (window.accessible_modal) {
+                            window.accessible_modal(
+                                '#' + $trigger.attr('id'),
+                                view.closeButtonSelector,
+                                '#' + view.$el.attr('id'),
+                                view.mainPageSelector
+                            );
+                        }
+                    });
+                },
+
+                handleTrigger: function(event) {
+                    var $trigger = $(event.target),
+                        courseName = $trigger.data('courseName'),
+                        courseNumber = $trigger.data('courseNumber'),
+                        isRefundable = $trigger.data('entitlementIsRefundable') && $trigger.data('entitlementIsRefundable').toLowerCase() === 'true',
+                        apiEndpoint = $trigger.data('entitlementApiEndpoint');
+
+                    this.resetModal();
+                    this.setTrackInfo(courseName, courseNumber);
+                    this.setRefundInfo(isRefundable);
+                    this.setSubmitData(apiEndpoint);
+
+                    if (window.edx && window.edx.dashboard && window.edx.dashboard.dropdown) {
+                        window.edx.dashboard.dropdown.toggleCourseActionsDropdownMenu(event);
+                        this.$el.css('position', 'fixed');
+                    }
+                },
+
+                handleSubmit: function() {
+                    var apiEndpoint = this.$submitButton.data('entitlementApiEndpoint');
+
+                    if (apiEndpoint === undefined) {
+                        this.setError(gettext('Error: cannot process unenrollment request.'));
```

Comment:
TODO: Need to confirm the correct text to set here.

Reply:
@catong could you please confirm that this is acceptable for an error message that will be displayed in the entitlement unenrollment modal when something goes wrong setting up the API request?

Reply:
This error should only happen if there's an error in the code that set up the data needed to make the API request, or if the user manipulates that data manually.

Reply:
@amangano-edx Can we change this to something in the active voice, and possibly include an action (Would "Try again" make sense?) Something like: "Your unenrollment request could not be processed. Try again later."

Reply:
Changed.